instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -21,10 +22,15 @@
 from opentelemetry.instrumentation.asgi.package import _instruments
 from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
 from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.trace import Span
 from opentelemetry.util.http import get_excluded_urls
 
 _excluded_urls = get_excluded_urls("STARLETTE")
 
+_ServerRequestHookT = typing.Optional[typing.Callable[[Span, dict], None]]
+_ClientRequestHookT = typing.Optional[typing.Callable[[Span, dict], None]]
+_ClientResponseHookT = typing.Optional[typing.Callable[[Span, dict], None]]
```

Comment:
Wondering if it makes sense to move these to the base instrumentation code?

Reply:
Imo they shouldn't because each instrumentation can have different types for the hooks e.g. Urllib has the following types. Starlette and FastAPI have the same types as they both use aiohttp 
```
_RequestHookT = typing.Optional[
     typing.Callable[[Span, urllib3.connectionpool.HTTPConnectionPool], None]
 ]
 _ResponseHookT = typing.Optional[
     typing.Callable[
         [
             Span,
             urllib3.connectionpool.HTTPConnectionPool,
             urllib3.response.HTTPResponse,
         ],
         None,
     ]
 ] 
```